gdnative-core/src/core_types/variant.rs

```diff
@@ -696,9 +696,14 @@ impl Variant {
         }
     }
 
+    /// Get a reference to a `godot-rust` Variant from a raw sys::pointer.
+    ///
+    /// # Safety
+    ///
+    /// The pointer must be a valid pointer to a `sys::godot_variant`.
     #[inline]
-    pub(crate) fn cast_ref<'l>(ptr: *const sys::godot_variant) -> &'l Variant {
-        unsafe { &*(ptr as *const variant::Variant) }
+    pub unsafe fn cast_ref<'l>(ptr: *const sys::godot_variant) -> &'l Variant {
```

Comment:
I think this should continue to be `pub(crate)` or at least `#[doc(hidden)]`? `gdnative-sys` is supposed to become an internal dependency going forward, so we don't want to expose `sys` types in the public interface.

Reply:
Just an artifact from the first try of splitting the nativescript things into a separate crate, good catch!